From 4147957a5eec57ec7a2a416dca74c3c0299a3432 Mon Sep 17 00:00:00 2001
From: Luca Boccassi <bluca@debian.org>
Date: Sun, 13 Jan 2019 13:08:10 +0000
Subject: [PATCH] Problem: test_security_zap fails on architectures that
 disallow unaligned pointer access

Solution: use memcpy instead of doing pointer arithmetics with casting
and dereferencing to fix the error on sparc64
---
 tests/testutil_security.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/testutil_security.hpp b/tests/testutil_security.hpp
index 90999118c..437bfb298 100644
--- a/tests/testutil_security.hpp
+++ b/tests/testutil_security.hpp
@@ -345,7 +345,7 @@ static int get_monitor_event_internal (void *monitor_,
     uint8_t *data = (uint8_t *) zmq_msg_data (&msg);
     uint16_t event = *(uint16_t *) (data);
     if (value_)
-        *value_ = *(uint32_t *) (data + 2);
+        memcpy (value_, data + 2, sizeof (uint32_t));
 
     //  Second frame in message contains event address
     zmq_msg_init (&msg);
